.github/workflows/release.yml

```diff
@@ -0,0 +1,152 @@
+        runtime: ${{ fromJSON(needs.enumerate.outputs.runtime) }}
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v3
+
+      - name: Cache target dir
+        uses: actions/cache@v3
+        with:
+          path: "${{ github.workspace }}/${{ matrix.runtime[0] }}/${{ matrix.runtime[1] }}/target"
+          key: srtool-target-${{ matrix.runtime[0] }}-${{ matrix.runtime[1] }}-${{ github.sha }}
+          restore-keys: |
+            srtool-target-${{ matrix.runtime[0] }}-${{ matrix.runtime[1] }}-
+            srtool-target-${{ matrix.runtime[0] }}-
+
+      - name: Build ${{ matrix.runtime[1] }}
+        id: srtool_build
+        uses: chevdor/srtool-actions@v0.7.0
+        with:
+          chain: ${{ matrix.runtime[1] }}
+          runtime_dir: "${{ matrix.runtime[0] }}/${{ matrix.runtime[1] }}"
+
+      - name: Store ${{ matrix.runtime[1] }} srtool digest to disk
+        run: |
+          echo '${{ steps.srtool_build.outputs.json }}' | jq > ${{ matrix.runtime[1] }}_srtool_output.json
+
+      - name: Upload ${{ matrix.runtime[1] }} srtool json
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ matrix.runtime[1] }}-srtool-json
+          path: ${{ matrix.runtime[1] }}_srtool_output.json
+
+      - name: Upload  ${{ matrix.runtime[1] }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ matrix.runtime[1] }}
+          path: |
+            ${{ steps.srtool_build.outputs.wasm_compressed }}
+
+  publish-draft-release:
+    runs-on: ubuntu-latest
+    needs:
+      - build-runtimes
+    outputs:
+      release_url: ${{ steps.create-release.outputs.html_url }}
+      asset_upload_url: ${{ steps.create-release.outputs.upload_url }}
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v3
+
+      - name: Download srtool json output
+        uses: actions/download-artifact@v3
+
+      - name: Archive context output
+        uses: actions/upload-artifact@v3
+        with:
+          name: release-notes-context
+          path: |
+            **/*_srtool_output.json
+
+      - name: Add runtime info to changelog
+        run: |
+          >>CHANGELOG echo "## Runtime info"
+          for JSON in $(find . -name '*_srtool_output.json'); do
+            C() { >>CHANGELOG echo $@; }
+            WASM() { <${JSON} jq -r .runtimes.compact.subwasm.$1; }
+
+            C "### $(WASM core_version.specName)"
+            C '```'
+            C "🏋️ Runtime Size:           $(WASM size) bytes"
+            C "🔥 Core Version:           $(WASM core_version.specName)-$(WASM core_version.specVersion) $(WASM core_version.implName)-$(WASM core_version.implVersion).tx$(WASM core_version.transactionVersion).au$(WASM core_version.authoringVersion)"
+            C "🎁 Metadata version:       V$(WASM metadata_version)"
+            C "🗳️ system.setCode hash:    $(WASM proposal_hash)"
+            C "🗳️ authorizeUpgrade hash:  $(WASM parachain_authorize_upgrade_hash)"
```

Comment:
IMO these two fields are not really the purpose of srtool. They often get wrapped in other calls/instructions like `with_weight` or `Transact {..}`, so nobody sees this in a proposal anyway. They are also sensitive to changes in transaction version.

The Blake2 hash of the code itself should be enough. Beyond that it is the proposer's job to apply the code correctly (and OpenGov's to reject it if not applied correctly).